src/views/Home.vue

```diff
@@ -1,18 +1,90 @@
 <template>
   <div class="home">
-    <img alt="Vue logo" src="../assets/logo.png">
-    <HelloWorld msg="Welcome to Your Vue.js App"/>
+    <h1>ToDoリスト</h1>
+    <label><input type="radio" value="すべて" v-model="filter" />すべて</label>
+    <label><input type="radio" value="作業中" v-model="filter" />作業中</label>
+    <label><input type="radio" value="完了" v-model="filter" />完了</label>
+    <table>
+      <thead>
+        <tr>
+          <th>ID</th>
+          <th>コメント</th>
+          <th>状態</th>
+          <th></th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="obj in filterTodos" :key="obj.id">
+          <td>{{ obj.id }}</td>
+          <td>{{ obj.task }}</td>
+          <td>
+            <button @click="statusTask(obj.id)">{{ obj.status }}</button>
+          </td>
+          <td>
+            <button @click="deleteTask(obj.id)">削除</button>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+    <h2>新規タスクの追加</h2>
+    <input type="text" id="task" v-model="inputTask" ref="focusThis" />
+    <button id="add" @click="addTask">追加</button>
   </div>
 </template>
 
 <script>
-// @ is an alias to /src
-import HelloWorld from '@/components/HelloWorld.vue'
+import definition from '@/helpers/definition';
 
 export default {
-  name: 'Home',
-  components: {
-    HelloWorld
-  }
-}
+  name: 'home',
+  data() {
+    return {
+      inputTask: '',
+      todos: [],
+      id: '',
+      task: '',
+      status: '',
+      deleteFlag: false,
+      filter: 'すべて',
+    };
+  },
+  methods: {
+    addTask() {
+      const obj = definition.addTask({
+        todos: this.todos,
+        task: this.inputTask,
+      });
+      this.todos = obj.addTodos;
+      this.inputTask = '';
+      this.$refs.focusThis.focus();
+    },
+    deleteTask(index) {
+      const obj = definition.deleteTask({
+        todos: this.todos,
+        _index: index,
+      });
+      this.todos = obj.deleteTodos;
+    },
+    statusTask(index) {
+      if (this.todos[index].status === '作業中') {
+        this.todos[index].status = '完了';
+      } else if (this.todos[index].status === '完了') {
+        this.todos[index].status = '作業中';
+      }
+    },
+  },
+  computed: {
+    filterTodos() {
+      let todos = [];
+      if (this.filter === 'すべて') {
+        todos = this.todos;
+      } else if (this.filter === '作業中') {
+        todos = this.todos.filter((todo) => todo.status === '作業中');
```

Comment:
filterは配列返却するものですので、以下のように記述するとtodosを定義する必要がなくなりコードもスッキリしますので、参考にして下さい。

```suggestion
        return this.todos.filter((todo) => todo.status === '作業中');
```